docker/release/config/opensearch-dashboards/opensearch-dashboards-docker-entrypoint.sh

```diff
@@ -160,8 +160,12 @@ function setupSecurityDashboardsPlugin {
         if [ "$DISABLE_SECURITY_DASHBOARDS_PLUGIN" = "true" ]; then
             echo "Disabling OpenSearch Security Dashboards Plugin"
             ./bin/opensearch-dashboards-plugin remove securityDashboards
-            cat $OPENSEARCH_DASHBOARDS_HOME/config/opensearch_dashboards.yml | sed "/^opensearch_security/d" | tee $OPENSEARCH_DASHBOARDS_HOME/config/opensearch_dashboards.yml
-            cat $OPENSEARCH_DASHBOARDS_HOME/config/opensearch_dashboards.yml | sed "s/https/http/g" | tee $OPENSEARCH_DASHBOARDS_HOME/config/opensearch_dashboards.yml
+
+            # Remove all security related parameters as well as changing HTTPS to HTTP
+            # Temporary fix before security-dashboards plugin implement a parameter to disable the plugin entirely
+            # https://github.com/opensearch-project/security-dashboards-plugin/issues/896
+            NEW_CONFIG=`cat $OPENSEARCH_DASHBOARDS_HOME/config/opensearch_dashboards.yml | sed "/^opensearch_security/d" | sed "s/https/http/g"`
```

Comment:
Note, I am very paranoid about this change

How do we know that `var=cat | sed | sed` `echo $var > ` is any more / less reliable than what we did in the past?

Reply:
It is more reliable because commands in a shell script are executed synchronously in sequence.

When you run something like this:
```
cmd1
cmd2
```

it is same as to run:
```
cmd1; cmd2
```

In previous iteration we are running cat/sed/tee in one single line/command, which cause the race condition to happen, as `tee` can set the file to empty before `cat` complete the reading of the file, since cat read in block size, thus cause the race condition to happen:
https://unix.stackexchange.com/questions/245499/how-does-cat-know-the-optimum-block-size-to-use

In new iteration, we allow cat and sed to complete processing of the entire file and save to the variable, before writing the content back to the file.

Since they are two different command, the 1st one will always complete, set the return code to 0, before the second command even start.

Thanks.

Reply:
Thanks for the detailed explaination!